cmd/commands/runtime.go

```diff
@@ -287,6 +287,12 @@ func runtimeInstallCommandPreRunHandler(cmd *cobra.Command, opts *RuntimeInstall
 		return err
 	}
 
+	opts.InsCloneOpts.Provider, err = inferProviderFromRepo(opts.InsCloneOpts.Repo)
```

Comment:
why do we infer provider always? what if --provider is used?

Reply:
i added a function that will infer the provider from the repo in an early stage, and if it can't it will fail. so this will always be with a valid value

Reply:
I don't understand. what if i use `cf runtime install foobar --provider github --repo https://some-git-hub.com/foo/bar`. Why is that not okay?